### PR TITLE
Arrays and loops done

### PR DESCRIPTION
**Pull request by AndreevSV**
Task with arrays and loops done

### CODE REVIEW

File: src/js/arrays_loops.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import { contacts } from "../data/contacts.js";
```
Comment (Avi-E-Koenig) [inline src/js/arrays_loops.js:1]: - good clean code in file!

File: src/js/arrays_loops.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+console.log(contacts);
```
Comment (Avi-E-Koenig) [inline src/js/arrays_loops.js]: remove unneeded consoles when handing in your work (exposes data,impacts performance)

File: src/js/arrays_loops.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+// Array Order Methods
+
+// 1.	Write a function that takes the array of contacts and returns a new array sorted by last name.
+
+const contactsArray = contacts.results;
+
+function sortedByLastName(contactsArray) {
```
Comment (Avi-E-Koenig) [inline src/js/arrays_loops.js]: note you are mutating the source here.
use 
```js
const cloneArr = structuredClone(array)
```

File: src/js/arrays_loops.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    }
+    return 0;
+  });
+}
+
+console.log(sortedByLastName(contactsArray));
+
+// 2.	Write a function to return the array of contacts in reverse order. Do not use the built-in reverse() method.
+function reverseOrderArray(contactsArray) {
+  let contactObject = {};
```
Comment (Avi-E-Koenig) [inline src/js/arrays_loops.js]: You are using a declaration here outside the needed scope
the declaration is also not needed.

File: src/js/arrays_loops.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    reversedOrderArray.push(contactObject);
+  }
+  return reversedOrderArray;
+}
+
+console.log(reverseOrderArray(contactsArray));
+
+// 3.	Write a function that returns the first 5 contacts from the sorted list (by last name).
+function returnFirstFiveSortedContacts(contactsArray) {
+  const sortedByLastNameArray = sortedByLastName(contactsArray);
```
Comment (Avi-E-Koenig) [inline src/js/arrays_loops.js]: good solution!

File: src/js/arrays_loops.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  return sortedByLastNameArray.slice(0, 5);
+}
+
+console.log(returnFirstFiveSortedContacts(contactsArray));
+
+// 4.	Create a function that returns an array of all unique first names. No duplicates should be present.
+function returnAllUniqueFirstNames(contactsArray) {
+  let firstNamesArray = [];
```
Comment (Avi-E-Koenig) [inline src/js/arrays_loops.js]: complexity of solution is too great.

please come up with more efficient solution.

File: src/js/arrays_loops.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+// 1.	Write a loop that iterates through the array and logs each contact's email to the console.
+function logEmails(contactsArray) {
+  contactsArray.forEach((contact) => console.log(contact.email));
+}
+
+logEmails(contactsArray);
+
+// 2.	 Write a function that takes an ID as a parameter and returns the contact with that ID.
+function findContactById(uuid) {
+  return contactsArray.find((contact) => contact.login.uuid == uuid);
```
Comment (Avi-E-Koenig) [inline src/js/arrays_loops.js]: nice!

File: src/js/arrays_loops.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+console.log(findContactById("2eb053db-195a-43c2-b371-5ad1109f7b71"));
+
+// 3.	Create a function that counts how many contacts are from a specific country. The country should be a parameter of the function.
+function countCountries(country) {
+  let counter = 0;
```
Comment (Avi-E-Koenig) [inline src/js/arrays_loops.js]: good use of loop
can you use array method here instead ?

File: src/js/arrays_loops.js [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    }
+  }
+  return counter;
+}
+
+let country = "Ukraine";
+console.log(`Contacts from country ${country}: `, countCountries(country));
+
+// 4.	Write a function that returns a new array of contacts that are within a given age range, e.g., 25 to 35 years old.
+function returnContactsWithAgeRange(contactsArray, ageMin, ageMax) {
```
Comment (Avi-E-Koenig) [inline src/js/arrays_loops.js]: good implementation.
use map and provide an alternative solution

**Review by Avi-E-Koenig — CHANGES_REQUESTED** [review]

**Comment by AndreevSV** [comment]
@Darkmift  Avi, please check my work after your supervision.

**Pull request merged**